Programming Course Draft 4.R

```diff
@@ -0,0 +1,433 @@
+
+    # turns the data into RR values
+
+    x <- as.data.frame(diff(imported_data[ ,1]))
+
+    # creating a vector that transforms RR values to milliseconds
+
+    time_vector <- vector()
+
+    for (value in 1:nrow(x)) {
+
+      newsum <- sum(x[1:value, ])
+
+      time_vector <- append(time_vector, newsum)
+    }
+
+    # millisecond to minutes
+
+    minutes_vector <- time_vector / 60000
+
+    # then add this to existing dataframe, so that the user can index time
+
+    data <- cbind(minutes_vector, x)
+
+  } else if (input == "RR" && segment == TRUE) {
+
+    x <- as.data.frame(imported_data[ ,1])
+
+    # create a new vector that converts RR values to time
+
+    time_vector <- vector()
+
+    for (value in 1:nrow(x)) {
+
+      newsum <- sum(x[1:value, ])
+
+      time_vector <- append(time_vector, newsum)
+    }
+
+    # millisecond to minutes
+
+    minutes_vector <- time_vector / 60000
+
+    # then add this to existing dataframe, so that the user can index time
+
+    df1_plus_time <- cbind(minutes_vector, x)
+
+    # referencing the row number based on the minimum value that the user inputs
+
+    min_row <- which(grepl(min, df1_plus_time[ ,2]))
+
+    min_row <- min_row[1]
+
+    # referencing the row number based on the maximum value that the user inputs
+
+    max_row <- which(grepl(max, df1_plus_time[ ,2]))
+
+    max_row <- tail(max_row)
+
+    max_row <- max_row[6]
+
+    # narrowing down the data frame so that it only includes the RR values
+
+    # within the user's specified time frame
+
+    data <- df1_plus_time[min_row:max_row, 1:2]
+
+  } else if (input == "MS" && segment == TRUE) {
+
+    x <- as.data.frame(diff(imported_data[ ,1]))
+
+    time_vector <- vector()
+
+    for (value in 1:nrow(x)) {
+
+      newsum <- sum(x[1:value, ])
+
+      time_vector <- append(time_vector, newsum)
+
+    }
+
+
+    # millisecond to minutes
+
+    minutes_vector <- time_vector / 60000
+
+    # then add this to existing dataframe, so that the user can index time
+
+    df1_plus_time <- cbind(minutes_vector, x)
+
+    # referencing the row number based on the minimum value that the user inputs
+
+    min_row <- which(grepl(min, df1_plus_time[ ,2]))
+
+    min_row <- min_row[1]
+
+    # referencing the row number based on the maximum value that the user inputs
+
+    max_row <- which(grepl(max, df1_plus_time[ ,2]))
+
+    max_row <- tail(max_row)
+
+    max_row <- max_row[6]
+
+    # narrowing down the data frame so that it only includes the RR values
+
+    # within the user's specified time frame
+
+    data <- df1_plus_time[min_row:max_row, 1:2]
+
+  }
+
+
+  # warning messages for incorrect data input
+
+  last_minute <- tail(data[,1])
+
+  last_minute <- last_minute[0]
+
+  if (last_minute < .5) {
+
+    warning("This software is not appropriate for heart rate
+            segments smaller than 30 seconds.")
+
+  }
+
+  if (segment = TRUE) {
+
+    if (min > max) {
```

Comment:
from the error message it looks like it should be 
`if (min + 30 > max)`
but im not sure